libbeat/autodiscover/autodiscover_test.go

```diff
@@ -173,19 +165,40 @@ func TestAutodiscover(t *testing.T) {
 	assert.True(t, runners[0].started)
 	assert.False(t, runners[0].stopped)
 
-	// Test stop event
+	// Test stop/start
 	eventBus.Publish(bus.Event{
 		"stop": true,
 		"meta": common.MapStr{
 			"foo": "baz",
 		},
 	})
+	eventBus.Publish(bus.Event{
+		"start": true,
+		"meta": common.MapStr{
+			"foo": "baz",
+		},
+	})
 	time.Sleep(10 * time.Millisecond)
 	runners = adapter.Runners()
-	assert.Equal(t, len(runners), 1)
-	assert.Equal(t, runners[0].meta.Get()["foo"], "baz")
-	assert.True(t, runners[0].started)
+	assert.Equal(t, len(runners), 2)
 	assert.True(t, runners[0].stopped)
+	assert.Equal(t, runners[1].meta.Get()["foo"], "baz")
+	assert.True(t, runners[1].started)
+	assert.False(t, runners[1].stopped)
+
+	// Test stop event
+	eventBus.Publish(bus.Event{
+		"stop": true,
+		"meta": common.MapStr{
+			"foo": "baz",
+		},
+	})
+	time.Sleep(10 * time.Millisecond)
```

Comment:
Let's hope it's enough on slow machines :-)

Reply:
I don't learn the lesson :), let see if jenkins like it, I'll can add some back-off code if not

Reply:
It's more a question of how often Jenkins will like it 😆